### ISSUE TITLE
Support Squarespace events calendar using json

### ISSUE THREAD

**Issue by kimadactyl** [geeksforsocialchange/PlaceCal#1500]
## User story

As a partner using Squarespace, I want to list events on PlaceCal using the default events calendar, in order that I can import them without any extra effort

## Acceptance criteria

- [x] ~We have a test Squarespace account that contains some events we can test with~ - we would need to pay for this so skipping, VCR for the given partner will have to do
- [x] Squarespace is a supported import source
- [ ] We have a handbook page that describes how to import Squarespace

## Implementation notes & questions

This is a request from [VFD](https://www.vfdalston.com/) who have confirmed this is what they use but are unable to share the password. Individual events have ics entries but the total feed doesn't (that I can see). Some URL hacking gives this page which seems the most promising source: https://www.vfdalston.com/our-events

## Implementation plan

- [Grab the json feed for page data](https://www.vfdalston.com/our-events?format=json)
- Extract event info from the "upcoming" json response noting there noting that the startDate is in unixtime

**Comment by kimadactyl** [comment]
Just to note that obviously the simplest solution for this will be if there is a hidden ical link somewhere in the config or in a random route!

**Comment by kimadactyl** [comment]
OK so on investigation

- There is no global iCal feed - boo
- There is a [JSON feed](https://www.vfdalston.com/our-events?format=json) and [rss feed](https://www.vfdalston.com/our-events?format=rss) which includes event info - but NOT the event date/time 😬 
- We could in theory use the url from there to load the rest of the event info - but this info is massively incomplete (See below for example) - compare to [the page itself](https://www.vfdalston.com/our-events/2022/6/20/transmissions-5l3da-dhw7k-tka72-8lpf4-fpcgk-y6a3w-5zckt-2jhtw-x2s9z-62l4l-mjkrg-8p8dj-473ln-w7jnp).
- However, this info is on the original json feed so we could glue the two together - is this a life we want to live though?

```ics
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Squarespace Inc/Squarespace 6//v6//EN
BEGIN:VEVENT
UID:630ddbada8c3a937c937075c@squarespace.com
DTSTAMP:20221019T155121Z
DTSTART:20221024T180000Z
DTEND:20221024T213000Z
SUMMARY:TRANSMISSIONS presents TOIL &amp\; TROUBLE: Possession (1981)
END:VEVENT
END:VCALENDAR
```

In summary, technically possible but sounds messy.

**Comment by kimadactyl** [comment]
@erbridge will leave this to you to triage - i think this isn't happening for transdim launch? sad, i thought this was going to be an easy win via ical or rss but having to use a weird amalgam of both makes me think its a Big Job

**Comment by kimadactyl** [comment]
OH! I missed that there is a unix startime [in the json feed](https://www.vfdalston.com/our-events?format=json) - that'll do!! I think this makes it way more doable?

<img width="658" alt="Screenshot 2022-10-24 at 17 33 46" src="https://user-images.githubusercontent.com/511734/197578402-ee64613e-8e51-45e2-a05a-511fe136a3bc.png">

**Comment by ivan-kocienski-gfsc** [comment]
Just reading through the comments and squarespace documentation and this to me feels like it could be tricky: how do we detect a squarespace page? the URLs are completely non-uniform and will likely be given to us in HTML

The fact it can come to us in JSON is fine though (i'd rather use JSON than ical).

**Comment by kimadactyl** [comment]
confirmed working!

<img width="1218" alt="Screenshot 2022-10-31 at 11 41 55" src="https://user-images.githubusercontent.com/511734/199000053-c52cc593-0418-4ee5-96fc-7a084c4e59a4.png">
<img width="810" alt="Screenshot 2022-10-31 at 11 42 02" src="https://user-images.githubusercontent.com/511734/199000046-ad8279ae-18a8-46f5-b036-83e3eb04f321.png">
